app/workers/pull_request_monitor.rb

```diff
@@ -10,12 +10,28 @@ def perform
     if !first_unique_worker?
       logger.info "#{self.class} is already running, skipping"
     else
-      Repo.includes(:branches).each do |repo|
-        # TODO: Need a better check for repos that *can* have PRs
-        next unless repo.upstream_user
+      Repo.includes(:branches).each { |repo| process_repo(repo) }
+    end
+  end
 
-        RepoProcessor.process(repo)
+  def process_repo(repo)
```

Comment:
This is the guts of the purpose of the PullRequestMonitor.  Otherwise you'd need a method on the Repo model called `process_for_pull_request_monitor`, which feels wrong.